lib/apollo-client.ts

```diff
@@ -59,57 +59,59 @@ const getCustomAgent = () => {
 };
 
 const serverSideFetch = async (url, options: { headers?: any; agent?: any; body?: string } = {}) => {
-  const nodeFetch = require('node-fetch');
-
-  options.agent = getCustomAgent();
-
-  // Add headers to help the API identify origin of requests
-  options.headers = options.headers || {};
-  options.headers['oc-env'] = process.env.OC_ENV;
-  options.headers['oc-secret'] = process.env.OC_SECRET;
-  options.headers['oc-application'] = process.env.OC_APPLICATION;
-  options.headers['user-agent'] = 'opencollective-frontend/1.0 node-fetch/1.0';
-
-  // Start benchmarking if the request is server side
-  const start = process.hrtime.bigint();
-
-  const result = await nodeFetch(url, options);
-
-  // Complete benchmark measure and log
-  if (process.env.GRAPHQL_BENCHMARK) {
-    const end = process.hrtime.bigint();
-    const executionTime = Math.round(Number(end - start) / 1000000);
-    const apiExecutionTime = result.headers.get('Execution-Time');
-    const graphqlCache = result.headers.get('GraphQL-Cache');
-    const latencyTime = apiExecutionTime ? executionTime - Number(apiExecutionTime) : null;
-    const body = JSON.parse(options.body);
-    if (body.operationName || body.variables) {
-      const pickList = [
-        'CollectiveId',
-        'collectiveSlug',
-        'CollectiveSlug',
-        'id',
-        'ledgacyId',
-        'legacyExpenseId',
-        'slug',
-        'term',
-        'tierId',
-      ];
-      const operationName = body.operationName || 'anonymous GraphQL query';
-      const variables = pick(body.variables, pickList) || {};
-      // eslint-disable-next-line no-console
-      console.log(
-        '-> Fetched',
-        operationName,
-        variables,
-        executionTime ? `in ${executionTime}ms` : '',
-        latencyTime ? `latency=${latencyTime}ms` : '',
-        graphqlCache ? `GraphQL Cache ${graphqlCache}` : '',
-      );
+  if (typeof window === 'undefined') {
```

Comment:
I like avoiding indentation change. And return early are always nice.

Reply:
I had tried an early return, but did not get the same results, I though that maybe webpack was not able to optimize. I will give it another shot.

Reply:
Interesting, good to know